src/arch/src/aarch64/regs.rs

```diff
@@ -114,31 +122,47 @@ macro_rules! arm64_sys_reg {
 // https://elixir.bootlin.com/linux/v4.20.17/source/arch/arm64/include/asm/sysreg.h#L135
 arm64_sys_reg!(MPIDR_EL1, 3, 0, 0, 0, 5);
 
-/// Configure core registers for a given CPU.
+/// Configure relevant boot registers for a given vCPU.
 ///
 /// # Arguments
 ///
 /// * `vcpu` - Structure for the VCPU that holds the VCPU's fd.
 /// * `cpu_id` - Index of current vcpu.
 /// * `boot_ip` - Starting instruction pointer.
 /// * `mem` - Reserved DRAM for current VM.
-pub fn setup_regs(vcpu: &VcpuFd, cpu_id: u8, boot_ip: u64, mem: &GuestMemoryMmap) -> Result<()> {
+pub fn setup_boot_regs(
+    vcpu: &VcpuFd,
+    cpu_id: u8,
+    boot_ip: u64,
+    mem: &GuestMemoryMmap,
+) -> Result<()> {
+    let kreg_off = offset__of!(kvm_regs, regs);
+
     // Get the register index of the PSTATE (Processor State) register.
-    vcpu.set_one_reg(arm64_core_reg!(pstate), PSTATE_FAULT_BITS_64)
-        .map_err(Error::SetCoreRegister)?;
+    let offset = offset__of!(user_pt_regs, pstate) + kreg_off;
+    vcpu.set_one_reg(
+        arm64_core_reg_id!(KVM_REG_SIZE_U64, offset),
+        PSTATE_FAULT_BITS_64,
+    )
+    .map_err(Error::SetCoreRegister)?;
 
     // Other vCPUs are powered off initially awaiting PSCI wakeup.
     if cpu_id == 0 {
         // Setting the PC (Processor Counter) to the current program address (kernel address).
-        vcpu.set_one_reg(arm64_core_reg!(pc), boot_ip)
+        let offset = offset__of!(user_pt_regs, pc) + kreg_off;
+        vcpu.set_one_reg(arm64_core_reg_id!(KVM_REG_SIZE_U64, offset), boot_ip as u64)
```

Comment:
The previous code was more readable, now I have a hard time figuring out the register which is on the line above. Can this be improved ?

Reply:
Maybe rename `offset` to a more verbose `pc_offset` (and respectively `pstate_offset` etc)?

Reply:
@aghecenco I applied your suggestion, @sandreim take a look!